src/components/cc-orga-member-list/cc-orga-member-list.smart.js

```diff
@@ -0,0 +1,252 @@
+      const currentUser = component.members.value.find((member) => member.isCurrentUser);
+      const isCurrentUserManager = currentUser.role === 'MANAGER';
+      const isMemberToUpdateAdmin = memberToUpdateRole === 'ADMIN';
+
+      return isCurrentUserManager && isMemberToUpdateAdmin;
+    }
+
+    function updateAuthorisations (role) {
+      const hasAdminRights = role === 'ADMIN' || role === 'MANAGER';
+
+      updateComponent('authorisations', {
+        invite: hasAdminRights,
+        edit: hasAdminRights,
+        delete: hasAdminRights,
+      });
+    }
+
+    function updateMember (memberId, callback) {
+      updateComponent('members', (members) => {
+        const member = members.value.find((member) => member.id === memberId);
+        if (member != null) {
+          callback(member);
+        }
+      });
+    }
+
+    onEvent('cc-orga-member-list:leave', ({ id }) => {
+      updateMember(id, (member) => {
+        member.state = 'deleting';
+      });
+
+      updateComponent('members', (members) => {
+        members.dangerZoneState = 'leaving';
+      });
+
+      deleteMember({ apiConfig, ownerId, id })
+        .then(() => {
+          notifySuccess(component, i18n('cc-orga-member-list.leave.success'));
+          updateComponent('members', (members) => {
+            members.value = members.value.filter((member) => member.id !== id);
+            members.state = 'loaded';
+            members.dangerZoneState = 'idle';
+          });
+          updateAuthorisations();
+          const leaveEvent = new Event('orga-member-leave-success');
+          window.dispatchEvent(leaveEvent);
+        })
+        .catch((error) => {
+          console.error(error);
+          notifyError(component, i18n('cc-orga-member-list.leave.error'));
+          updateMember(id, (member) => {
+            member.state = 'loaded';
+          });
+          updateComponent('members', (members) => {
+            members.dangerZoneState = 'idle';
+          });
+        });
+    });
+
+    onEvent('cc-orga-member-list:invite', ({ email, role }) => {
+
+      updateComponent('inviteMemberForm', (inviteMemberForm) => {
+        // Note: the UI component already resets the errors and sets the field values
+        inviteMemberForm.state = 'inviting';
+      });
+
+      postNewMember({ apiConfig, ownerId, email, role })
+        .then(() => {
+          notifySuccess(component, i18n('cc-orga-member-list.invite.submit.success', { userEmail: email }));
+          updateComponent('inviteMemberForm', CcOrgaMemberList.INIT_INVITE_FORM_STATE);
+        })
+        .catch((error) => {
+          console.error(error);
+          if (error.id === UNAUTHORISED_ADMIN_ADDITION) {
+            notifyError(component, i18n('cc-orga-member-list.error.unauthorised.text'), i18n('cc-orga-member-list.error.unauthorised.heading'));
+          }
+          else if (error.message === 'You have performed that request too much.') {
```

Comment:
move `'You have performed that request too much.'` into a constant near `UNAUTHORISED_ADMIN_ADDITION`